### PR TITLE
Add mirror env variable when set by the user

### PR DESCRIPTION
**Pull request by tgodzik**
This can be found https://github.com/coursier/coursier/blob/master/modules/paths/src/main/java/coursier/paths/Mirror.java#L26

Tested  it out and seems to work (URLs go replaced by my string)

Will create VS Code PR soon.

### CODE REVIEW

File: src/detectLaunchConfigurationChanges.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export function detectLaunchConfigurationChanges(
       "serverProperties",
       "javaHome",
       "customRepositories",
+      "coursierMirror",
```
Comment (kpodsiad) [inline src/detectLaunchConfigurationChanges.ts:20]: Can we extract these keys as constants and use them in vscode?

**Review by kpodsiad — APPROVED** [review]

File: src/detectLaunchConfigurationChanges.ts (hunk 1 above)
Comment (tgodzik) [reply src/detectLaunchConfigurationChanges.ts:20]: In a couple of places yes, but unfortunately not in `package.json` which would be most useful

Comment (kpodsiad) [reply src/detectLaunchConfigurationChanges.ts:20]: That's true, but at least we can reduce 3 possible sources of errors (config options names) to just 2

**Review by ckipp01 — APPROVED** [review]

File: src/interfaces/UserConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * These are some of the common user configuration options used in Metals and needed for the editors.
+ *
+ *  - https://scalameta.org/metals/docs/integrations/new-editor#metals-user-configuration
+ */
+export const UserConfiguration = {
```
Comment (kpodsiad) [inline src/interfaces/UserConfiguration.ts:6]: I think you can use enum, with it exporting type shouldn't be necessary

Comment (tgodzik) [reply src/interfaces/UserConfiguration.ts]: Changed!

**Pull request merged**